packages/seed/src/fixture-storage.ts

```diff
@@ -82,27 +99,45 @@ export class FixtureStorage<T> {
    * Creates a new seed fixtureStorage for managing seed fixtures.
    * @param uri Fixture uri
    */
-  constructor(public uri: string, public fileExtension?: string) {}
+  constructor(
+    readonly uri: string,
+    readonly fileExtension?: string,
+    readonly testScope?: TestScope
+  ) {}
 
   /**
    * Get fixture.
    * @param uri Fixture uri
    */
   get(): T {
-    const fixture = pathExistsSync(this.path)
-      ? readJsonSync(this.path)
-      : undefined
+    const fixture = pathExistsSync(this.path) && this.fs.read(this.path)
 
     if (!fixture) {
-      throw new SeedError('Http: fixture (seed) is missing.', this)
+      throw new SeedError(`Fixture (seed) is missing: ${this.path}.`, this)
     }
     if (this.expired) {
-      throw new SeedError('Http: fixture (seed) is expired.', this)
+      throw new SeedError(`Fixture (seed) is expired: ${this.path}.`, this)
+    }
+    if (this.testScope) {
+      this.testScope.callIndex += 1
     }
-
     return fixture
   }
 
+  private get fs() {
+    const jsonOpts = { spaces: 2 }
+    switch (this.fileExtension) {
```

Comment:
I do not like switch cases, because they tend to be a flood gate for overboarding complexity (hey, look at me, sounding like a medium.com article). Please do not use them. 

I would do something like this
```javascript
  const handleXMLorWSDL = { save: outputFileSync, read: readFileSync }
  const handleJSON = {
          read: readJsonSync,
          save: (path: string, val: any) => outputJsonSync(path, val, jsonOpts),
        }
  return isXMLorWSDL() ? handleXMLorWSDL : handleJSON

```

Reply:
i think an if statement is not a very good alternative here, as this will not be extendable, e.g.
````typescript
return isXMLorWSDL() ? isJson ? handleJSON : handleXMLorWSDL : handleCSV
````
will be very hard to read. How about this instead:
````typescript
const fileFormatFs = new Map<FixtureFileExtension, FsHandler>([
  ['.json', this.fsJson],
  ['.wsdl', this.fsXml],
  ['.xml', this.fsXml],
  ['.csv', ...]
])
return fileFormatFs.get(this.fileExtension) || this.fsJson
````
Note: same would work with a plain dictionary, but Map is easier to handle defaults